[Pranavi] iP

pra-navi opened the pull request:
# Sally

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

## Sally frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

## All you need to do is,
1. download it from here.
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

### And it is **FREE**!

## Features:
- [X] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

## If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
`System.out.printf("Yay, you are done!");`

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

iknowuknow committed:
Sally welcomes

iknowuknow committed:
Echo

iknowuknow committed:
add to list

iknowuknow committed:
mark as done

iknowuknow committed:
three types of tasks

iknowuknow committed:
Automated Text UI Testing

iknowuknow committed:
Exceptions

iknowuknow committed:
delete

iknowuknow committed:
Shorten methods

iknowuknow committed:
Save tasks to hard disk

pra-navi committed:
Merge pull request #1 from pra-navi/branch-level-7

Save tasks to hard disk

iknowuknow committed:
Convert time and date to uniform setting

iknowuknow committed:
Convert time and date to uniform setting

pra-navi committed:
Merge branch 'master' into branch-level-8

pra-navi committed:
Merge pull request #2 from pra-navi/branch-level-8

Convert time and date to uniform setting

iknowuknow committed:
Update the date and time from separate class to main

iknowuknow committed:
Adhere to OOP

iknowuknow committed:
Sort into packages

iknowuknow committed:
Set up gradle and JAR file

iknowuknow committed:
Merge branch 'add-gradle-support'

iknowuknow committed:
Create JUnit tests

iknowuknow committed:
Add javadocs

iknowuknow committed:
Adhere to coding standards

iknowuknow committed:
Add find-by-keyword function

pra-navi committed:
Merge pull request #3 from pra-navi/branch-A-CodingStandard

Adhere to coding standards

pra-navi committed:
Merge branch 'master' into branch-A-JavaDoc

pra-navi committed:
Merge pull request #4 from pra-navi/branch-A-JavaDoc

Add javadocs

pra-navi committed:
Merge branch 'master' into branch-Level-9

pra-navi committed:
Merge pull request #5 from pra-navi/branch-Level-9

Add find-by-keyword function

iknowuknow committed:
Add find test to test folder

iknowuknow committed:
Incorporate checkstyle

iknowuknow committed:
Merge remote-tracking branch 'origin/branch-level-7'

iknowuknow committed:
View first iteration of GUI, no images and one-sided inputs

trgao commented on src/main/java/sally/Task.java:
```diff
@@ -0,0 +1,140 @@
+ */
+public class Task {
+    protected String task;
+    protected boolean toBeDone;
```
Perhaps you can change the name of the boolean to isDone?

trgao commented on src/main/java/sally/Task.java:
```diff
@@ -0,0 +1,140 @@
+            LocalDateTime from = convertToDateTime(toGetFromTo[1]);
+            LocalDateTime to = convertToDateTime(toGetFromTo[2]);
+            newTask = new Event(task, from, to);
+
```
Perhaps you can remove this whitespace, as well as the one in the else block below?

trgao commented on src/main/java/sally/AddEventCommand.java:
```diff
@@ -0,0 +1,87 @@
+    private static LocalDateTime convertToDateTime(String input) {
+        String pattern1 = "\\d{4}-\\d{1,2}-\\d{1,2} \\d{4}";
+        String pattern2 = "\\d{4}-\\d{1,2}-\\d{1,2}";
+        String pattern3 = "\\d{4}";
```
Perhaps you can change these 3 strings into constants?

trgao commented on src/main/java/sally/Parser.java:
```diff
@@ -0,0 +1,44 @@
+            String[] parts = input.split(" ", 2);
+            String commandType = parts[0].toLowerCase();
+
+            if (commandType.equals("mark")) {
```
Perhaps you can use switch statement so that it looks cleaner?

trgao reviewed (commented):
LGTM, just very very minor things to comment on.

iknowuknow committed:
View second iteration of GUI, only repeated responses from Sally

iknowuknow committed:
Include greeting and goodbye message

iknowuknow committed:
Include commands into GUI

iknowuknow committed:
Add a separate Message file with all Strings to be shown to user

iknowuknow committed:
Flip the conversation between Sally and the User

pra-navi committed:
Merge pull request #7 from pra-navi/branch-level-10

branch-level-10

brandon-nam commented on src/main/java/sally/Task.java:
```diff
@@ -0,0 +1,140 @@
+     * @param fileString The string representation of the task from the file.
+     * @return The Task object created from the file string.
+     */
+    public static Task fromFileString(String fileString) {
```
Perhaps this method could be broken down to several sub methods?

brandon-nam commented on src/main/java/sally/Parser.java:
```diff
@@ -0,0 +1,44 @@
+        } else if (!input.contains(" ")) {
+            throw new SallyException("Sorry, I do not understand the command :-(");
+        } else {
+            String[] parts = input.split(" ", 2);
```
Perhaps could be more explanatory with the use of 2 by giving it a variable name?

brandon-nam commented on src/main/java/sally/Deadline.java:
```diff
@@ -0,0 +1,36 @@
+    @Override
+    public String toString() {
+        String formattedDate;
+        DateTimeFormatter outputFormatter = DateTimeFormatter.ofPattern("MMM dd yyyy, h:mm a");
```
Perhaps the date format could be replaced with a variable name for better readability and reusability?

brandon-nam commented on src/main/java/sally/AddEventCommand.java:
```diff
@@ -0,0 +1,89 @@
+        Matcher matcher1 = regexPattern1.matcher(input);
+        Matcher matcher2 = regexPattern2.matcher(input);
+        Matcher matcher3 = regexPattern3.matcher(input);
+
```
I like how you grouped relevant statements together and added spacing for readability :)

brandon-nam reviewed (commented):
LGTM overall! Just some minor changes recommendations :)

iknowuknow committed:
Include assertions to code

Some parameters of methods might be null which we can assume to be not null

So assert statements have been added to validate these assumptions later at runtime

iknowuknow committed:
Implement CodeQuality to current code

Some methods were updated to return String. Some boolean variable was not named according to the convention.

But these updates were not reflected in the javadocs.

Let's,
* update the javadocs to reflect the new return values
* rename the boolean variable

pra-navi committed:
Merge pull request #8 from pra-navi/branch-A-Assertions

Include assertions to code

pra-navi committed:
Merge pull request #9 from pra-navi/branch-A-CodeQuality

Implement CodeQuality to current code

iknowuknow committed:
Include the priority function

Now can tag each task with a priority.

The priority, as an enum, is marked as either low, medium, high or tbd.

The priority can be updated.

pra-navi committed:
Merge pull request #10 from pra-navi/branch-C-Priority

Include the priority function

Mahidharah commented on src/main/java/sally/Storage.java:
```diff
@@ -0,0 +1,75 @@
+     * @return The TaskList loaded from the file.
+     * @throws SallyException If there's an issue while loading the tasks.
+     */
+    public TaskList loadTasksFromFile() throws SallyException {
```
ArrowHead code

iknowuknow committed:
Add product image

iknowuknow committed:
Add product image

iknowuknow committed:
Update UserGuide

iknowuknow committed:
Update JUnit

iknowuknow committed:
Create JAR file

pra-navi committed:
Fix UG
